Makefile

```diff
@@ -0,0 +1,170 @@
+# Settings
+# --------
+
+build_dir:=.build
+deps_dir:=deps
+defn_dir:=$(build_dir)/defn
+k_submodule:=$(deps_dir)/k
+pandoc_tangle_submodule:=$(deps_dir)/pandoc-tangle
+k_bin:=$(k_submodule)/k-distribution/target/release/k/bin
+tangler:=$(pandoc_tangle_submodule)/tangle.lua
+wasm_submodule:=$(deps_dir)/wasm-semantics
+eei_submodule:=$(deps_dir)/eei-semantics
+
+LUA_PATH=$(pandoc_tangle_submodule)/?.lua;;
+export LUA_PATH
+
+.PHONY: all clean \
+        deps ocaml-deps haskell-deps \
+        defn defn-ocaml defn-java defn-haskell \
+        build build-ocaml defn-haskell build-haskell \
+        test test-execution test-simple test-prove test-klab-prove \
+        media presentations reports definition-deps
+
+all: build
+
+clean:
+	rm -rf $(build_dir)
+	git submodule update --init --recursive
+
+# Build Dependencies (K Submodule)
+# --------------------------------
+
+deps: $(k_submodule)/make.timestamp $(pandoc_tangle_submodule)/make.timestamp ocaml-deps definition-deps
+
+$(k_submodule)/make.timestamp:
+	git submodule update --init --recursive
+	cd $(k_submodule) && mvn package -DskipTests -Dllvm.backend.skip
+	touch $(k_submodule)/make.timestamp
+
+$(pandoc_tangle_submodule)/make.timestamp:
+	git submodule update --init -- $(pandoc_tangle_submodule)
+	touch $(pandoc_tangle_submodule)/make.timestamp
+
+ocaml-deps:
+	eval $$(opam config env) \
+	    opam install --yes mlgmp zarith uuidm
+
+# Building Definition
+# -------------------
```

Comment:
Importantly, this build system will not build the submodules for EEI and Wasm independently. It will just grab the definition files from those submodules.

* Pros: simplicity 
* Cons: integration testing isn't automatic when you make a change in a submodule; slower builds(?)